### PR TITLE
chore: bump postgres from 10 to 14

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY

Postgres 11.0 change log: https://www.postgresql.org/docs/release/11.0/
Postgres 12.0 change log: https://www.postgresql.org/docs/release/12.0/
Postgres 13.0 change log: https://www.postgresql.org/docs/release/12.0/
Postgres 14.0 change log: https://www.postgresql.org/docs/release/14.0/

Also updates helm postgres bitnami chart to 11.1.22, that sets postgres to 14.2:
https://github.com/bitnami/charts/blob/master/bitnami/postgresql/values.yaml#L90

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by villebro** [comment]
Two more things I think we should do:
- add docs about 14 being officially supported
- update `helm/superset/Chart.yaml`: https://github.com/apache/superset/blob/f06db796b5a609915d96b0a176f474d5142d9813/helm/superset/Chart.yaml#L27-L28

**Review by craig-rueda — APPROVED** [review]

File: docs/docs/installation/configuring-superset.mdx [hunk 1]
```diff
@@ -69,6 +69,32 @@ you can add the endpoints to `WTF_CSRF_EXEMPT_LIST`:
 WTF_CSRF_EXEMPT_LIST = [‘’]
 ```
 
+### Using a production metadata database
```
Comment (villebro) [inline docs/docs/installation/configuring-superset.mdx:72]: We've started to prefer the term "metastore", as "metadata database" is a bit of a tongue twister. But we can do a full update of that in a follow-up PR.

File: docs/docs/installation/configuring-superset.mdx [hunk 2]
```diff
@@ -69,6 +69,32 @@ you can add the endpoints to `WTF_CSRF_EXEMPT_LIST`:
...
+
+By default Superset is configured to use SQLite, it's a simple and fast way to get you started
+(no installation needed) but for production environments you should use a different database engine on
+a separate host or container.
```
Comment (villebro) [inline docs/docs/installation/configuring-superset.mdx:76]: Maybe we could split this up into two sentences:
```suggestion
By default Superset is configured to use SQLite, it's a simple and fast way to get you started
(no installation needed). But for production environments you should use a different database engine on
a separate host or container.
```

File: docs/docs/installation/configuring-superset.mdx [hunk 3]
```diff
@@ -69,6 +69,32 @@ you can add the endpoints to `WTF_CSRF_EXEMPT_LIST`:
...
+
+Superset supports the following database engines/versions:
+
+| Database Engine                                           | Supported Versions                |
+| --------------------------------------------------------- | --------------------------------- |
+| [MySQL](https://www.mysql.com/)                           | 5.X                               |
+| [PostgreSQL](https://www.postgresql.org/)                 | 10.X, 11.X, 12.X, 13.X, 14.X      |
```
Comment (villebro) [inline docs/docs/installation/configuring-superset.mdx:83]: Could we kinda recommend using Postgres by placing it on top in the list? Or maybe even explicitly recommend it?
```suggestion
Superset supports the following database engines/versions:

| Database Engine                                           | Supported Versions                |
| --------------------------------------------------------- | --------------------------------- |
| [PostgreSQL](https://www.postgresql.org/)                 | 10.X, 11.X, 12.X, 13.X, 14.X      |
| [MySQL](https://www.mysql.com/)                           | 5.X                               |
```

**Review by villebro — APPROVED** [review]
LGTM with a few non-blocking comments. And big thanks for adding the detailed docs - this piece of documentation has been sorely needed for a long time!

**Pull request merged**